src/main/java/io/stargate/sgv2/jsonapi/api/model/command/table/definition/indexes/RegularIndexDefinitionDesc.java

```diff
@@ -15,17 +30,117 @@
   TableDescConstants.IndexDefinitionDesc.OPTIONS
 })
 public record RegularIndexDefinitionDesc(
+
+    /*
+     * API regular column can be primitive or map/set/list collection type.
+     * CreateIndexCommand on collection column will have index function.
+     * User can specify or use the default index function.
+     * Details see {@link RegularIndexColumnDeserializer}
+     */
     @NotNull
-        @Schema(description = "Name of the column to index.", required = true)
+        @Schema(
+            description =
+                "Name of the column to index. (Optional: can also specify the index function for map column. E.G. {\"column\": {\"mapColumn\" : \"$keys\"}}",
+            required = true)
+        @JsonDeserialize(using = RegularIndexColumnDeserializer.class)
+        @JsonSerialize(using = RegularIndexColumnSerializer.class)
         @JsonProperty(TableDescConstants.IndexDefinitionDesc.COLUMN)
-        String column,
-    //
+        RegularIndexColumn column,
     @JsonInclude(JsonInclude.Include.NON_NULL)
         @Nullable
         @Schema(description = "Indexing options.", type = SchemaType.OBJECT)
         @JsonProperty(TableDescConstants.IndexDefinitionDesc.OPTIONS)
         RegularIndexDescOptions options)
-    implements IndexDefinitionDesc<RegularIndexDefinitionDesc.RegularIndexDescOptions> {
+    implements IndexDefinitionDesc<
+        RegularIndexDefinitionDesc.RegularIndexColumn,
+        RegularIndexDefinitionDesc.RegularIndexDescOptions> {
+
+  public record RegularIndexColumn(
+      @NotNull
+          @Schema(description = "Name of the column to index.", required = true)
+          @JsonProperty(TableDescConstants.IndexDefinitionDesc.COLUMN)
+          String columnName,
+      @Nullable
+          @Schema(description = "Index function for map/set/list column.")
```

Comment:
nit: lets hide the idea of a "index function" and this only applies to maps we are not going to expose the idea of functions for set and list. 

"Property of the map to index, either $keys or $values " something like that